composer/loggers/slack_logger.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Log metrics to slack, using Slack postMessage api."""
+import os
+import time
+from typing import Any, Callable, Dict, List, Optional, Set
+
+from composer.loggers.logger_destination import LoggerDestination
+from composer.utils import MissingConditionalImportError
+
+
+class SlackLogger(LoggerDestination):
+    """Log metrics to slack, using Slack's postMessage api - https://api.slack.com/methods/chat.postMessage.
+
+    First export 2 environment variable to use this logger.
+    1. SLACK_LOGGING_API_KEY: To get app credentials, follow tutorial here - https://api.slack.com/tutorials/tracks/posting-messages-with-curl?app_id_from_manifest=A053W1QCEF2.
+    2. SLACK_LOGGING_CHANNEL_ID: Channel id to send the message (Open slack channel in web browser to look this up).
+
+    Next write script to output metrics / hparams / traces to slack channel. See example below.
+
+    .. code-block:: python
+
+        trainer = Trainer(
+            model=mnist_model(num_classes=10),
+            train_dataloader=train_dataloader,
+            max_duration='2ep',
+            algorithms=[
+                LabelSmoothing(smoothing=0.1),
+                CutMix(alpha=1.0),
+                ChannelsLast(),
+            ],
+            loggers=[
+                SlackLogger(
+                    formatter_func=(lambda data, **kwargs: [{
+                        'type': 'section',
+                        'text': {
+                            'type': 'mrkdwn',
+                            'text': f'*{k}:* {v}'
+                        }
+                    } for k, v in data.items()]),
+                    include_keys={'loss/train/total'},
+                    interval_in_seconds=1
+                ),
+            ],
+        )
+
+        trainer.fit()
+
+    Args:
+        formatter_func ((...) -> Any | None): A formatter function that returns list of blocks to be sent to slack.
+        include_keys (Set[str]): A set of metric/logs/traces keys to include in the message. If None, all keys are included.
```

Comment:
I think we might want to tweak the defaults here. I don't think by default all keys should be included. It might be too easy for a user to just pollute their slack channel accidentally. We might want to make this a required argument. If the user passes in a sequence, then those are the keys logged and if they pass in 'all' or '*' or something then all keys are logged